src/clique_parser.erl

```diff
@@ -340,20 +408,50 @@ validate_invalid_flags_test() ->
     Spec = spec(),
     Args = [],
     Node = "dev2@127.0.0.1",
-    InvalidFlags = [{'some-flag', Node},
+    InvalidFlags = [{"some-flag", Node},
                     {$b, Node},
                     {$a, undefined}],
     [?assertMatch({error, _}, validate({Spec, Args, [F], []})) || F <- InvalidFlags].
 
 validate_valid_args_test() ->
     Spec = spec(),
-    Args = [{sample_size, "5"}],
+    Args = [{"sample_size", "5"}],
     {undefined, ConvertedArgs, [], []} = validate({Spec, Args, [], []}),
     ?assertEqual(ConvertedArgs, [{sample_size, 5}]).
 
 validate_invalid_args_test() ->
     Spec = spec(),
-    InvalidArgs = [{key, "value"}, {sample_size, "ayo"}],
+    InvalidArgs = [{"key", "value"}, {"sample_size", "ayo"}],
     [?assertMatch({error, _}, validate({Spec, [A], [], []})) || A <- InvalidArgs].
 
+
+arg_datatype_test() ->
+    Spec = dt_validate_spec(),
+    ValidArg = [{"sample_size", "10"}],
+    {undefined, ConvertedArgs, [], []} = validate({Spec, ValidArg, [], []}),
+    ?assertEqual(ConvertedArgs, [{sample_size, 10}]),
+
+    InvalidTypeArg = [{"sample_size", "A"}],
+    ?assertMatch({error, {conversion, _}}, validate({Spec, InvalidTypeArg, [], []})).
```

Comment:
This test is failing for me. The actual error returned by cuttlefish appears to be ```{error,"\"A\" can't be converted to an integer"}```, which doesn't match the {error, {conversion, _}} pattern.